src/utils/axiosInterceptors.ts

```diff
@@ -37,22 +46,25 @@ const fetchCachedResponse = async (request: any): Promise<any> => {
     return request;
   }
 
-  const response = await cache.match(cacheKey);
-  if (response) {
+  const cachedResponse = await cache.match(cacheKey);
+  if (cachedResponse) {
+    if (!hasCachedResponseExpired(cachedResponse)) {
+      return request;
+    }
```

Comment:
Nitpicking: can we negate the check so we can get rid of `if (cachedResponse) {` and lower the nesting level?

Something like this:
```
if (!cachedResponse || !hasCachedResponseExpired(cachedResponse)) {
  return request;
}
...
```

Reply:
Thanks, have gone with what you suggested

Reply:
Not exactly what I meant, sorry if I wasn't clear enough... `if (cachedResponse) {` still hides a very long block of code. One could avoid this by doing:
```
if (!cachedResponse || !hasCachedResponseExpired(cachedResponse)) {
  return request;
}

request.adapter = async () => {
  ...
};
return request;
```